Modules/IO/NIFTI/test/itkNiftiImageIOTest14.cxx

```diff
@@ -0,0 +1,109 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkNiftiImageIOTest.h"
+#include "itkTestingMacros.h"
+
+// Test writing NIFTI xyzt_units field - should be 10 (mm, sec)
+int
+itkNiftiImageIOTest14(int argc, char * argv[])
+{
+
+  if (argc > 1)
+  {
+    char * testdir = argv[1];
+    itksys::SystemTools::ChangeDirectory(testdir);
+  }
+  else
+  {
+    return EXIT_FAILURE;
+  }
+
+  constexpr unsigned int Dimension = 3;
+
+  using PixelType = float;
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  ImageType::Pointer image;
+
+  // fill out an image and write it as nifti.
+  const char * filename = "xyzt_unit_check.nii";
+  using OutputImageType = itk::Image<PixelType, Dimension>;
+  typename OutputImageType::RegionType region;
+  typename OutputImageType::IndexType  start;
+  typename OutputImageType::SizeType   size;
+  size[0] = 8;
+  size[1] = 8;
+  size[2] = 4;
+  region.SetSize(size);
+  region.SetIndex(start);
```

Comment:
So again, I would suggest to remove the `region.SetIndex(start)` call. But if you _really_ want that call it like that, you should make sure that `start` is properly initialized. (You can do so, just by adding a pair of curly braces to its declaration: `typename OutputImageType::IndexType start{};`) But I would suggest to simply remove the `start` variable.

Reply:
Thanks, I just want a small blank image as I'm only testing the header. I got rid of start entirely and also changed Allocate() to AllocateInitialized(), does this look better?

```
  // fill out an image and write it as nifti.
  const char * filename = "xyzt_unit_check.nii";
  using OutputImageType = itk::Image<PixelType, Dimension>;
  typename OutputImageType::RegionType region;
  typename OutputImageType::SizeType   size;
  size[0] = 8;
  size[1] = 8;
  size[2] = 4;
  region.SetSize(size);
  auto outputimage = OutputImageType::New();
  outputimage->SetRegions(region);
  outputimage->AllocateInitialized();
```

Reply:
Thanks @cookpa I guess with Allocate(), "xyzt_unit_check.nii" would just contain arbitrary pixel values, which would probably be OK to your test. It's just the C++ defines read-access to uninitialized data as "undefined behavior", meaning that _anything_ might have happened.

BTW, `itk::Size<N>` is a simple aggregate struct, which supports list-initialization, for example as follows:

    typename OutputImageType::SizeType   size{ 8, 8, 4 };

So that you don't really need to do `size[0] = 8; size[1] = 8; size[2] = 4;`. But that's just a matter of style.

Reply:
I ended up not using the code to create a blank image, because I added test images to capture more failures. Now I test this by writing one of them back out and checking that the units are correct.